lib/Command/CreateClient.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * @author Ilja Neumann <ineumann@owncloud.com>
+ * @copyright Copyright (c) 2017, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ */
+
+namespace OCA\OAuth2\Command;
+
+use Doctrine\DBAL\Exception\UniqueConstraintViolationException;
+use OCA\OAuth2\Db\Client;
+use OCA\OAuth2\Db\ClientMapper;
+use OCA\OAuth2\Utilities;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+
+class CreateClient extends \Symfony\Component\Console\Command\Command {
+
+	public function __construct($name) {
+		parent::__construct($name);
+	}
+
+	protected function configure() {
+		$this
+			->setName('oauth2:create-client')
+			->setDescription('Create a OAuth2 client (admin action)')
+			->addArgument('name', InputArgument::REQUIRED, 'Name of the client')
+			->addArgument('redirect-uri', InputArgument::REQUIRED, 'Redirect Url')
+			->addOption('identifier', 'I', InputOption::VALUE_OPTIONAL, "Random generated, set manually for testing only!")
```

Comment:
Somehow it's not working as optional for me: https://asciinema.org/a/uiCKTvla2tQQ8NNUUXcZMRFct

```
$ occ oauth2:create-client test http://localhost:*  

  [Doctrine\DBAL\Exception\NotNullConstraintViolationException]
  An exception occurred while executing 'INSERT INTO "oc_oauth2_clients"("secret","name","redirect_uri","allow_subdomains") VALUES(?,?,?,?)' with params ["ooWibiRzcxgPzThxVhBChbCdlitsMIoQrVG2X8wYc20A2HgDxH2Pz3HklQ9fXYBF", "test_branding", "http:\/\/localhost:*", false]
  :
  SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: oc_oauth2_clients.identifier
```